discovery-infra/test_infra/controllers/node_controllers/libvirt_controller.py

```diff
@@ -384,21 +418,53 @@ def _wait_till_domain_has_ips(self, domain, timeout=360, interval=5):
             expected_exceptions=Exception
         )
 
+    @staticmethod
+    def _clean_domain_os_boot_data(node_xml):
+        os_element = node_xml.getElementsByTagName('os')[0]
+
+        for el in os_element.getElementsByTagName('boot'):
+            dev = el.getAttribute('dev')
+            if dev in ['cdrom', 'hd']:
+                os_element.removeChild(el)
+            else:
+                raise ValueError(f'Found unexpected boot device: \'{dev}\'')
+
+        for disk in node_xml.getElementsByTagName('disk'):
+            for boot in disk.getElementsByTagName('boot'):
+                disk.removeChild(boot)
+
+    def set_per_device_boot_order(self, node_name, key: Callable[[Disk], int]):
```

Comment:
Maybe change the function name to clarify it's going to reboot the node (`set_per_device_boot_order_and_reboot` or something), or alternatively perform the reboot separately

Reply:
The reboot is a technical detail not the propose of the method

Reply:
It's a pretty surprising technical detail, I wouldn't expect a function that sets the boot order to reboot my node

Reply:
I agree that we don't expect a reboot but it's a side effect and we want this encapsulation. 
Our code was written in general (no strong connection to the libvirt implementation) and this behavior might be changed in the next libvirt version or for another controller (which isn't supported yet)